dpe/src/context.rs

```diff
@@ -150,70 +150,173 @@ pub(crate) struct ChildToRootIter<'a> {
     idx: usize,
     contexts: &'a [Context],
     done: bool,
+    count: usize,
 }
 
 impl ChildToRootIter<'_> {
     /// Create a new iterator that will start at the leaf and go to the root node.
-    pub fn new(leaf_idx: usize, contexts: &[Context]) -> Result<ChildToRootIter, DpeErrorCode> {
-        Ok(ChildToRootIter {
+    pub fn new(leaf_idx: usize, contexts: &[Context]) -> ChildToRootIter {
+        ChildToRootIter {
             idx: leaf_idx,
             contexts,
             done: false,
-        })
+            count: 0,
+        }
     }
 }
 
 impl<'a> Iterator for ChildToRootIter<'a> {
-    type Item = &'a Context;
+    type Item = Result<&'a Context, DpeErrorCode>;
 
-    fn next(&mut self) -> Option<&'a Context> {
+    fn next(&mut self) -> Option<Result<&'a Context, DpeErrorCode>> {
         if self.done {
             return None;
         }
+        if self.count >= MAX_HANDLES {
+            return Some(Err(DpeErrorCode::MaxTcis));
+        }
 
         let context = &self.contexts[self.idx];
+
+        // Check if context is valid.
+        const MAX_IDX: u8 = (MAX_HANDLES - 1) as u8;
+        let valid_parent_idx = matches!(context.parent_idx, 0..=MAX_IDX | Context::ROOT_INDEX);
+        if !valid_parent_idx || context.state == ContextState::Inactive {
+            return Some(Err(DpeErrorCode::InternalError));
```

Comment:
Same question here regarding infinite iteration.

Reply:
Done

Reply:
This doesn't look done to me. I don't see any mutation before returning which means another call to next should return the same error.

Reply:
Oh weird somehow that line didn't get added to my commit. Should be good now.